### PR TITLE
Optimizing month_to_season function

### PR DESCRIPTION
**Pull request by hCraker**
The curent version of month_to_season in [climatology.py](https://github.com/NCAR/geocat-comp/blob/main/src/geocat/comp/climatology.py) calculates a seasonal mean by calculating the mean for all four seasons versus just the user-specified season of interest. This results in unnecessary computations.

In an effort to improve the efficiency of this function, I have implemented a version that filters the data before computation.

These are the results of the comparison test when ran on a Casper computation node with 1 CPU with 10 GB of memory.
```
---Test 6 million data points---
Old Method
0.11941519472748041 seconds
New Method
0.05952004995197058 seconds

Change:  50.15705489757617 %


---Test 60 million data points---
Old Method
0.7728856382891536 seconds
New Method
0.6625110507011414 seconds

Change:  14.280843390017639 %


---Test 600 million data points---
Old Method
10.698792199604213 seconds
New Method
8.26200267020613 seconds

Change:  22.776304875687085 %
```
The average performance improvement for 600 million data points averaged over 30 trials with randomized data was 25.56%.

### CODE REVIEW

File: src/geocat/comp/climatology.py [hunk 1]
```diff
@@ -284,39 +284,40 @@ def month_to_season(
...
-    },
-                                 loffset=month_offset).mean()
-
-    # Filter just the desired season, and trim to the desired time range.
-    compute_dset = dset_seasons.sel({
-        time_coord_name: dset_seasons[time_coord_name].dt.month == season_sel
-    }).sel({time_coord_name: slice(start_date, end_date)})
-    return compute_dset
+    # Filter data to only contain the months of interest
+    month_1st = dset.sel({time_coord_name: dset[time_coord_name].dt.month == month1})
```
Comment (dcherian) [inline src/geocat/comp/climatology.py:308]:
```suggestion
    data_filter = dset.sel({time_coord_name: dset[time_coord_name].dt.month.isin(months)})
```

where `months=(month1,month2,month3)`

Comment (hCraker) [reply src/geocat/comp/climatology.py:308]: That is a much more concise way of doing this. I'm going to double check how it affects performance before accepting this suggestion.

Comment (hCraker) [reply src/geocat/comp/climatology.py:308]: Thank you so much for this suggestion! I didn't realize that `.isin()` existed, and now the optimized version runs about twice as fast as the old one

**Comment by hCraker** [comment]
Performance update:
```
---Test 6 million data points---
Old Method
0.5723645305261016 seconds
New Method
0.033852167427539825 seconds

Change:  94.08555813260759 %


---Test 60 million data points---
Old Method
0.5565323093906045 seconds
New Method
0.2899643750861287 seconds

Change:  47.89801594742345 %


---Test 600 million data points---
Old Method
6.964924789965153 seconds
New Method
3.12193295173347 seconds
```
Average over 30 trials for 600 million data points: 57.21% faster

File: src/geocat/comp/climatology.py (hunk 1 above)
Comment (dcherian) [reply src/geocat/comp/climatology.py]: happy to help!

**Comment by hCraker** [comment]
@pilotchute All of the checks pass now. Could you review this when you have a moment? Thanks!

**Review by pilotchute — APPROVED** [review]
Performance improvements look great. Thanks!

**Pull request merged**